# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/DNSRequestChild.cpp
# Commit: 26c5be98ea21
# Full Hash: 26c5be98ea21c8c331cefe9649e3eb53197cd821
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-08-26 03:24:37
# Regressor Bug: 1652667
# File Overlap Count: 1
# Description:
#   Bug 1652667 - P2: Telemetry for whether a connection is succeeded by using an HTTPSSVC record r=valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D87857
# ==============================================================================

diff -r 96bd98319c62 -r 26c5be98ea21 netwerk/dns/DNSRequestChild.cpp
--- a/netwerk/dns/DNSRequestChild.cpp	Tue Aug 25 17:29:32 2020 +0000
+++ b/netwerk/dns/DNSRequestChild.cpp	Tue Aug 25 17:29:31 2020 +0000
@@ -190,6 +190,7 @@
   virtual ~ChildDNSByTypeRecord() = default;
 
   TypeRecordResultType mResults = AsVariant(mozilla::Nothing());
+  bool mAllRecordsExcluded = false;
 };
 
 NS_IMPL_ISUPPORTS(ChildDNSByTypeRecord, nsIDNSByTypeRecord, nsIDNSRecord,
@@ -197,7 +198,7 @@
 
 ChildDNSByTypeRecord::ChildDNSByTypeRecord(const TypeRecordResultType& reply,
                                            const nsACString& aHost)
-    : DNSHTTPSSVCRecordBase(aHost) {
+    : DNSHTTPSSVCRecordBase(aHost), mAllRecordsExcluded(false) {
   mResults = reply;
 }
 
@@ -258,8 +259,8 @@
   }
 
   auto& results = mResults.as<TypeRecordHTTPSSVC>();
-  nsCOMPtr<nsISVCBRecord> result =
-      GetServiceModeRecordInternal(aNoHttp2, aNoHttp3, results);
+  nsCOMPtr<nsISVCBRecord> result = GetServiceModeRecordInternal(
+      aNoHttp2, aNoHttp3, results, mAllRecordsExcluded);
   if (!result) {
     return NS_ERROR_NOT_AVAILABLE;
   }
@@ -282,6 +283,18 @@
 }
 
 NS_IMETHODIMP
+ChildDNSByTypeRecord::GetAllRecordsExcluded(bool* aResult) {
+  NS_ENSURE_ARG(aResult);
+
+  if (!mResults.is<TypeRecordHTTPSSVC>()) {
+    return NS_ERROR_NOT_AVAILABLE;
+  }
+
+  *aResult = mAllRecordsExcluded;
+  return NS_OK;
+}
+
+NS_IMETHODIMP
 ChildDNSByTypeRecord::GetResults(mozilla::net::TypeRecordResultType* aResults) {
   *aResults = mResults;
   return NS_OK;